app/controllers/katello/api/v2/api_controller.rb

```diff
@@ -216,5 +217,44 @@ def check_disconnected
             "is prohibited. If you would like to change this, please update the content setting 'Disconnected mode'."
       fail HttpErrors::BadRequest, _(msg) if Setting[:content_disconnected]
     end
+
+    def filtered_associations
+      nil
+    end
+
+    def filter_association_ids
+      if filtered_associations
+        find_param_arrays.each do |key_path|
+          if (model_class = filtered_associations.with_indifferent_access.dig(*key_path))
+            filtered_ids = model_class.readable.where(:id => params.dig(*key_path)).pluck(:id)
+            replace_params_value(key_path, filtered_ids)
+          else
+            fail _("Unfiltered params array: %s") % key_path
+          end
+        end
+      end
+    end
+
+    def replace_params_value(key_path, value, hash = params)
+      if key_path.length == 1
+        hash[key_path.first] = value
+      else
+        replace_params_value(key_path[1..-1], value, hash[key_path[0]])
```

Comment:
Two recursive methods in one PR!! :lollipop:

Reply:
![all](http://py.prof.ninja/class10/building.gif)